arrows/pdal/register_algorithms.cxx

```diff
@@ -0,0 +1,42 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief PDAL algorithm registration implementation
+
+#include <arrows/pdal/kwiver_algo_pdal_plugin_export.h>
+#include <vital/algo/algorithm_factory.h>
+
+#include <arrows/pdal/pointcloud_io.h>
+
+namespace kwiver {
+namespace arrows {
+namespace pdal {
+
+extern "C"
+KWIVER_ALGO_PDAL_PLUGIN_EXPORT
+void
+register_factories( kwiver::vital::plugin_loader& vpm )
+{
+  static auto const module_name = std::string( "arrows.pdal" );
+  if (vpm.is_module_loaded( module_name ) )
+  {
+    return;
+  }
+
+  // add factory               implementation-name       type-to-create
+  auto fact = vpm.ADD_ALGORITHM( "pdal", kwiver::arrows::pdal::pointcloud_io );
```

Comment:
This is the old deprecated way of registering algorithms.  The new way pulls this information from that `PLUGIN_INFO` call in the class itself and has simpler syntax.  See example here:
https://github.com/Kitware/kwiver/blob/master/arrows/mvg/register_algorithms.cxx#L55